### PR TITLE
Zephyr implementation missing functions

### PR DESCRIPTION
**Pull request by bdilly**
Required to have samples working on this platform.

Signed-off-by: Bruno Dilly <bruno.dilly@intel.com>

### CODE REVIEW

File: src/lib/comms/sol-network-impl-zephyr.c [hunk 1]
```diff
@@ -191,3 +191,33 @@ sol_network_link_get_name(const struct sol_network_link *link)
 {
     return NULL;
 }
+
+SOL_API int
+sol_network_hostname_pending_cancel(
+    struct sol_network_hostname_pending *handle)
+{
+    return -ENOTSUP;
```
Comment (cabelitos) [inline src/lib/comms/sol-network-impl-zephyr.c]: Maybe we should return -ENOSYS here as well, just like the other functions

Comment (cabelitos) [inline src/lib/comms/sol-network-impl-zephyr.c]: And also add a warn

File: src/lib/comms/sol-network-impl-zephyr.c [hunk 2]
```diff
@@ -191,3 +191,33 @@ sol_network_link_get_name(const struct sol_network_link *link)
...
+    return -ENOSYS;
+}
+
+SOL_API struct sol_network_hostname_pending *
+sol_network_get_hostname_address_info(const struct sol_str_slice hostname,
+    enum sol_network_family family, void (*host_info_cb)(void *data,
+    const struct sol_str_slice host, const struct sol_vector *addrs_list),
+    const void *data)
+{
+    return NULL;
```
Comment (cabelitos) [inline src/lib/comms/sol-network-impl-zephyr.c]: We should add a warn in here

Comment (bdilly) [inline src/lib/comms/sol-network-impl-zephyr.c]: We could do it, at least a INF.
I was following the RIOT implementation and it's used by some samples. So we're kind of expecting it to fail, so not so sure if it should warn.

**Comment by bdilly** [comment]
merged with proposed changes

**Pull request merged**